### PR TITLE
add autotranslate feature

### PR DESCRIPTION
**Pull request by domcross**
This PR adds autotranslate feature for utterances that are not in English.
Translation is done by Google Translate using pypi package "translators".

Works really well in German:

- wie hoch ist der Eifelturm / how tall is the eiffel tower
- wann wurde der Taj Mahal gebaut / when was the taj mahal built
- wer ist bundeskanzler von Deutschland / who is chancelor of Germany
- Was ist Kernspaltung / what is nuclear fission
- Wer erfand das Auto / who invented the automobile
- Wer ist Barack Obama / who is barack obama

### CODE REVIEW

**Comment by krisgesling** [comment]
Hey Dom, this looks really useful. 

My concern is that they state MIT license on their PyPI page, however they have modified it to include text that seems to be in contradiction to that license:
> Warning: Prohibition of Commercial Use!
This module is designed to help students and individuals with translation services.
For commercial use, please purchase API services from translation suppliers.

Whilst we are an open source project, we also sell devices. If it were a Community contributed Skill being submitted to the Marketplace I would allow it. However this being an official Mycroft Skill and one that is installed by default on all devices, I don't think we can use this library.

@penrods any thoughts?

**Comment by krisgesling** [comment]
Oh side note, is NodeJS really required for this?

**Comment by domcross** [comment]
Commercial use: to my understanding the licence warning is not concerning the `translators` package itself but the commercial use of 3rd-party APIs like Google-Translate? There are several other pypi-packages with similar functionality that in the end use the same 3rd-party-API and they seem not to care about that issue - I have found `mtranslate` with unmodified MIT licence

NodeJS: The `translators` library uses PyExecJS which requires a JS-engine. NodeJS is one of several options that is supported. The other options are Google-V8 (don't be evil), PhantomJS (support suspended) and Nashorn (by Oracle, I think they just caused a big mess with the JDK8 support). For me NodeJS was the next and best option. 
`mtranslate` does not require any JS-Engine

tl;dr: replaced package `translators` with `mtranslate` because of nicer licence and requirements

**Comment by domcross** [comment]
@krisgesling I have changed to another translation library `mtranslate` which should resolve the licence issue

**Comment by krisgesling** [comment]
Looks great, thanks Dom!
